### PR TITLE
Configurable opacity and mouse buttons

### PR DESCRIPTION
**Pull request by cdump**
I've made two changes:
1) configurable opacity for resize canvas, I prefer 0.75 in my configs
2) configurable mouse button for move/resize events - now it's possible to configure this spoon like https://awesomewm.org/
```
sky = SkyRocket:new({
  opacity = 0.75,
  moveModifiers = {'cmd'},
  moveMouseButton = 'left',
  resizeModifiers = {'cmd'},
  resizeMouseButton = 'right',
})
```

### CODE REVIEW

**Comment by dbalatero** [comment]
Hi @cdump! Thanks for the PR. I am currently on vacation but I can look at this when I'm back at a computer after Dec 25!

**Comment by cdump** [comment]
Hi @dbalatero! Did you get a chance to look into this?

File: init.lua [hunk 1]
```diff
@@ -59,19 +59,34 @@ end
...
 --   })
 --
+local function buttonNameToEventType(name)
+    if name == 'left' then
+        return hs.eventtap.event.types.leftMouseDown
+    end
+    if name == 'right' then
+        return hs.eventtap.event.types.rightMouseDown
+    end
+    error('only "left" and "right" mouse button supported, got ' .. name)
```
Comment (dbalatero) [inline init.lua]: **chore**: Please update this error message to include the name of the parameter that they misconfigured (`resizeMouseButton` or `moveMouseButton`)

File: init.lua [hunk 2]
```diff
@@ -59,19 +59,34 @@ end
...
 --   resizer = SkyRocket:new({
+--     opacity = 0.3,
 --     moveModifiers = {'cmd', 'shift'},
+--     moveMouseButton = 'left',
 --     resizeModifiers = {'ctrl', 'shift'}
+--     resizeMouseButton = 'left',
 --   })
 --
+local function buttonNameToEventType(name)
+    if name == 'left' then
```
Comment (dbalatero) [inline init.lua]: **chore** Please match the local indentation style in this file (2 spaces), thanks!

File: init.lua [hunk 3]
```diff
@@ -80,17 +95,26 @@ function SkyRocket:new(options)
   self.__index = self
 
   resizer.clickHandler = hs.eventtap.new(
-    { hs.eventtap.event.types.leftMouseDown },
+    {
+        hs.eventtap.event.types.leftMouseDown,
```
Comment (dbalatero) [inline init.lua]: **chore** 2 space indent, fix all cases (I won't comment this everywhere, please find all instances and fix)

File: init.lua [hunk 4]
```diff
@@ -200,9 +224,10 @@ function SkyRocket:handleClick()
     if self.dragging then return true end
 
     local flags = event:getFlags()
+    local etype = event:getType()
```
Comment (dbalatero) [inline init.lua]: **nit** Prefer expressive and clear variable names like `eventType`. Abbreviating `event` to `e` obscures what this variable is.

File: init.lua [hunk 5]
```diff
@@ -59,19 +59,34 @@ end
...
+end
+
 function SkyRocket:new(options)
   options = options or {}
 
   local resizer = {
     disabledApps = tableToMap(options.disabledApps or {}),
     dragging = false,
     dragType = nil,
+    moveStartMouseEvent = buttonNameToEventType(options.moveMouseButton or 'left'),
```
Comment (dbalatero) [inline init.lua]: **praise** Thanks for keeping the defaults the same and not breaking past users.

**Review by dbalatero — CHANGES_REQUESTED** [review]
Hi @cdump! Thanks for sending this in, it's pretty clear and easy to follow.

A few comments for you to address, then I will test the branch locally and if it seems good I will approve the PR and merge!

**Comment by cdump** [comment]
@dbalatero thanks for review, I've fixed all

**Review by dbalatero — APPROVED** [review]

**Comment by dbalatero** [comment]
Awesome thank you! Tested it and still works on my end.

**Pull request merged**